src/Stream/DynamicMemoryWriter.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "BidirectionalWriter.h"
+#include "MemoryReader.h"
+#include <cstddef>
+#include <cstdint>
+#include <vector>
+
+namespace Stream
+{
+	class DynamicMemoryWriter : public BidirectionalWriter
```

Comment:
Recommend adding a short description right above the class declaration explaining what the DynamicMemoryWriter does and why. Doesn't need to get into specifics and details, but I don't feel like DynamicMemoryWriter by itself fully explains the purpose of the class (and I don't know that it can be explained succinctly by just the name of the class?)

Reply:
Good idea about leaving some documentation.